elastic/elastic-agent
# Only run providers that are referenced in the policy

## What does this PR do?



This updates the Elastic Agent to only run composable providers if they are referenced in the policy.

## Why is it important?



This change provides the following benefits:

- Reduces the memory and load of the Elastic Agent, by only running the providers it references.
- No longer requires access to the kubernetes API or resources just to run an Elastic Agent (as long as its not using any ${kubernetes*` references in the policy).
- Fixes an issue where if no providers are running that a set of empty variables is passed to the coordinator.

## Checklist



- ~~[ ] I have added an integration test or an E2E test~~ (covered by unit tests)

## Disruptive User Impact



None. As it still runs all those that are referenced from the user.

## How to test this PR locally



Run the elastic-agent container with just the system integration and notice that it doesn't require access to the Kubernetes API any more.

## Related issues


- Closes #3609 
- Closes #4648
